spec/controllers/admin_controller_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe AdminController, type: :controller do
+  describe '#authenticate_admin' do
```

Comment:
It's for other controllers to inherit from. Application controller doesn't have any actions either.